keras_cv/models/task.py

```diff
@@ -112,57 +158,29 @@ def from_preset(
         ```
         """
 
-        if not cls.presets:
-            raise NotImplementedError(
-                "No presets have been created for this class."
+        # TODO: delete me!
+        if preset in cls.presets:
+            return cls._legacy_from_preset(
+                preset, load_weights, input_shape, **kwargs
             )
 
-        if preset not in cls.presets:
-            raise ValueError(
-                "`preset` must be one of "
-                f"""{", ".join(cls.presets)}. Received: {preset}."""
-            )
-
-        if load_weights and preset not in cls.presets_with_weights:
-            raise ValueError(
-                f"""Pretrained weights not available for preset "{preset}". """
-                "Set `load_weights=False` to use this preset or choose one of "
-                "the following presets with weights:"
-                f""" "{'", "'.join(cls.presets_with_weights)}"."""
-            )
+        preset_cls = check_preset_class(preset, (cls, cls.backbone_cls))
 
-        metadata = cls.presets[preset]
-        # Check if preset is backbone-only model
-        if preset in cls.backbone_presets:
-            backbone_cls = keras.saving.get_registered_object(
-                metadata["class_name"]
+        # Backbone case.
+        if preset_cls == cls.backbone_cls:
```

Comment:
I don't think this is what we want actually. I believe with CV (unlike NLP), the intent is to allow mixing and matching of backbones and tasks. So we should probably check the `preset_cls` is a subclass of `Backbone`, as I was mentioning earlier.

Reply:
Got it, I've updated to check if it's a subclass of Backbone